crates/typst-library/src/meta/target.rs

```diff
@@ -0,0 +1,87 @@
+use crate::prelude::*;
+use typst::export;
+
+/// Provides access to the target format, and associated information.
+///
+/// This allows show rules to only be applied in certain formats.
+/// ```example
+/// #set text(red) if target().is-pdf()
+/// ```
+///
+/// ### is-pdf()
+/// True when the target is a pdf, which is the default when the target is not specified and the extension isn't something else.
+///
+/// - returns: bool
+///
+/// ### is-svg()
+/// True when the target is a svg.
+///
+/// - returns: bool
+///
+/// ### is-png()
+/// True when the target is a png.
+///
+/// - returns: bool
+///
+/// ### is-query()
+/// True when the target is a query and the target isn't overridden.
+///
+/// - returns: bool
+///
+/// ### is-pageless()
+/// True when the target is a pageless format. HTML will be pageless, and queries are pageless.
+///
+/// - returns: bool
+///
+/// ### is-layouted()
+/// True when the target is a layouted format. PDF, SVG, and PNG are layouted.
+///
+/// - returns: bool
+///
+/// Display: Target
+/// Category: meta
+#[func]
+pub fn target(
+    /// The virtual machine.
+    vm: &mut Vm,
+) -> Target {
+    Target::new(vm.vt.world.target())
+}
+
+#[derive(Clone, PartialEq, Hash, Debug)]
+pub struct Target(export::Target);
```

Comment:
The `Debug` (and, thus, `repr()`) impl should probably just provide a string version of `Target` (e.g. `"pdf"`, `"png"`, `"svg"`, `"query"`).

Reply:
As I understand Debug, it's about internal representation

Reply:
> As I understand Debug, it's about internal representation

It's used by `repr` for user-facing value types, and thus doesn't normally expose the internal Rust structure (as is, you'd see something like `Target(Png)` or similar, which isn't consistent with other types).
https://github.com/typst/typst/blob/88bcb8769a07d1bad57b77fa9443b598fd1a6b61/crates/typst/src/eval/value.rs#L143-L146

Reply:
Fixed.